### PR TITLE
Expose custom http headers to MATLAB

### PR DESCRIPTION
**Pull request by stfnp**
This is my first attempt at exposing the custom http headers from Java-WebSocket to MatlabWebSocket (see issue #14 ).

The class WebSocketClient now has another optional parameter where a map of keys and values for the headers can be passed. For example, adding the authorization header `Authorization: Basic ZGl0dG86ZGl0dG8=` can be done like this:

    headers = containers.Map('Authorization', 'Basic ZGl0dG86ZGl0dG8=')
    client = SimpleClient(address, headers);

This pull request isn't yet finished (documentation, tests... ), but I wanted to get your feedback before continuing. In particular the following things:

* The Java constructor takes the http headers as a `Map<String, String>`, so I used maps in MATLAB too. But the interoperation is a bit awkward and the MATLAB maps don't seem to be very great. It would be possible to just use an array of strings, maybe that would be more convenient for users. Pythons [websocket_client](https://pypi.org/project/websocket_client/) for example does it like this.

* Is the argument parsing in WebSocketClient still correct? I haven't used this library with a keystore yet, so I can't confirm it. The tests still seem to work though.

* I built the `jar`-File with Java 11. Is this okay or should I downgrade (Java 8?) for more compatibility?

### CODE REVIEW

**Comment by jebej** [comment]
Thanks for the contribution! Is there a way I could test this feature on a public server somewhere?

1. From what I understand, it seems like the matlab `containers.Map` is automatically converted to the Java map, is that correct? If so, that seems like the simplest solution. I assume anything else would require some additional conversion code.

2. The argument parsing will not work as is, since the constructor still refers to the arguments `keyStore,storePassword,keyPassword`, and the input was changed to `varargin`.
In fact the parsing will end up being a bit complex in order to support all four combinations: ws+no headers, wss+no headers, ws+headers, & wss+header.

3. I'll build the jar file myself and push to this branch once everything works.

Thanks again!

File: src/WebSocketClient.m [hunk 1]
```diff
@@ -17,6 +17,7 @@
     
     properties (SetAccess = private)
         URI % The URI of the server
+        httpHeaders = containers.Map % Map of additional http headers
```
Comment (jebej) [inline src/WebSocketClient.m]: Could you capitalize this property (`HttpHeaders`)?

**Comment by jebej** [comment]
I see that you are in fact already converting the `containers.Map` object into a java `HashMap`. Then yes, a cell array with an even number of elements `{'key1','val1','key2','val2',...} `would also work fine.

**Comment by stfnp** [comment]
Oh right, the argument parsing was of course wrong... I made another commit and changed it to how I think it is correct. And I also did the change from map to a cell array with key value pairs. The usage could now look like this:

    headers = {'Authorization', 'Basic ZGl0dG86ZGl0dG8='}
    client = SimpleClient(address, headers);

Unfortunately I don't know a way to test it on a public server at the moment. We use your library (thank you for your great work, by the way!) to connect with [Eclipse Ditto](https://www.eclipse.org/ditto/index.html) (IoT framework). They do have a public [sandbox](https://www.eclipse.org/ditto/sandbox.html) instance, which would be great for testing, but it has been offline for a while now.

**Comment by stfnp** [comment]
The sandbox I mentioned is now online again, here is a little example program that you could use for testing:

    function main()
        DITTO_IP = 'ditto.eclipse.org';
        DITTO_PORT = '80';

        THING_FEATURE = '';
        THING_USR = 'demo1';
        THING_PWD = 'demo';

        address = get_address(DITTO_IP, DITTO_PORT, THING_FEATURE);
        headers = get_headers(THING_USR, THING_PWD);

        client = SimpleClient(address, headers);
        client.send('START-SEND-MESSAGES');
        client.send('START-SEND-EVENTS'); 

        pause();
    end

    function address = get_address(ip, port, feature)
        address = ['ws://', ip, ':', port, '/ws/1', feature];
    end

    function headers = get_headers(username, password)
        import matlab.net.base64encode;
        authorization = ['Basic ' base64encode([username ':' password])];
        headers = {'Authorization', authorization};
    end

**Comment by jebej** [comment]
Thanks, I made a few more changes and this all works now!

**Pull request merged**